ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -197,10 +197,12 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
                                         if (isArchivedWorkflow(workflow) && (globalDeleteArchived || !isWorkflowInCluster(workflow))) {
                                             services.workflows.deleteArchived(workflow.metadata.uid, workflow.metadata.namespace).catch(setError);
                                         }
-                                        navigation.goto(uiUrl(`workflows/${workflow.metadata.namespace}`));
-                                        // TODO: This is a temporary workaround so that the list of workflows
-                                        //  is correctly displayed. Workflow list page needs to be more responsive.
-                                        window.location.reload();
+                                        if (error === null) {
+                                            navigation.goto(uiUrl(`workflows/${workflow.metadata.namespace}`));
+                                            // TODO: This is a temporary workaround so that the list of workflows
+                                            //  is correctly displayed. Workflow list page needs to be more responsive.
+                                            window.location.reload();
+                                        }
```

Comment:
this `if` statement could be inverted to remove nesting. same with the `if (yes)` one above too

```suggestion
                                        if (error !== null) {
                                            return
                                        }
                                        navigation.goto(uiUrl(`workflows/${workflow.metadata.namespace}`));
                                         // TODO: This is a temporary workaround so that the list of workflows
                                         //  is correctly displayed. Workflow list page needs to be more responsive.
                                         window.location.reload();
```

Reply:
Refactored in #11711